--- a/arch/mips/cavium-octeon/executive/cvmx-helper-board.c
+++ b/arch/mips/cavium-octeon/executive/cvmx-helper-board.c
@@ -768,6 +768,7 @@ enum cvmx_helper_board_usb_clock_types _
 	case CVMX_BOARD_TYPE_NIC10E_66:
 	case CVMX_BOARD_TYPE_UBNT_E100:
 	case CVMX_BOARD_TYPE_CUST_DSR1000N:
+	case CVMX_BOARD_TYPE_CUST_NB5:
 		return USB_CLOCK_TYPE_CRYSTAL_12;
 	case CVMX_BOARD_TYPE_NIC10E:
 		return USB_CLOCK_TYPE_REF_12;
