Index: linux-4.4.14/arch/mips/pci/pci-octeon.c
===================================================================
--- linux-4.4.14.orig/arch/mips/pci/pci-octeon.c
+++ linux-4.4.14/arch/mips/pci/pci-octeon.c
@@ -216,6 +216,8 @@ const char *octeon_get_pci_interrupts(vo
 		return "AABCD";
 	case CVMX_BOARD_TYPE_CUST_DSR1000N:
 		return "CCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCC";
+	case CVMX_BOARD_TYPE_CUST_NB5:
+		return "ABDABAAAAAAAAAAAAAAAAAAAAAAAAAAA";
 	case CVMX_BOARD_TYPE_THUNDER:
 	case CVMX_BOARD_TYPE_EBH3000:
 	default: